pkg/api/types.go

```diff
@@ -1212,6 +1215,23 @@ type StorageOSPersistentVolumeSource struct {
 	SecretRef *ObjectReference
 }
 
+// HugePagesSource represents Linux HugeTlbPage https://www.kernel.org/doc/Documentation/vm/hugetlbpage.txt
+type HugePagesVolumeSource struct {
+	// Defaults to 2M
+	// +optional
+	PageSize string
+	// Defaults to PageSize
+	// The size option sets the maximum value of memory (huge pages).
```

Comment:
I can't rename it because there is a generated method Size(), so i will leave MaxSize as struct field but change json, and protobuf field name to size.